### ISSUE TITLE
Non-ASCII chars are not pct-encoded

### ISSUE THREAD

**Issue by C06A** [hal4j/uritemplate#8]
According to RFC-6570 the only unencoded characters are from unreserved set and in case of Reserved and Fragment Expansions -- also from reserved set.
I try to use Cyrillic chars in the values and they are not pct-encoded.

**Comment by ivan-gammel** [comment]
@C06A see linked pull request: I cannot reproduce the issue in unit tests. Could you please provide more details or better a test that would demonstrate the bug?

**Comment by C06A** [comment]
Here is the simple test:

```
    @Test
    public void simpleCyrilicTest() {
        String s = new URITemplate("{+cyrillic}")
                .expand( "cyrillic", "АБВГДЕйка")
                .toString();
        assertEquals("%D0%90%D0%91%D0%92%D0%93%D0%94%D0%95%D0%B9%D0%BA%D0%B0", s);
    }
```

which results in:

```
org.opentest4j.AssertionFailedError: 
Expected :%D0%90%D0%91%D0%92%D0%93%D0%94%D0%95%D0%B9%D0%BA%D0%B0
Actual   :АБВГДЕйка
<Click to see difference>
```

<img width="1749" alt="Screen Shot 2023-09-04 at 12 06 09 PM" src="https://github.com/hal4j/uritemplate/assets/13039234/d9575871-a757-4890-affe-f97e004dee9c">

**Comment by ivan-gammel** [comment]
Fixed in version 2.0.0
